### ISSUE TITLE
[Bug] `rich` text format doesn't show help (required/default/choices)

### ISSUE THREAD

**Issue by mrjsj** [BrianPugh/cyclopts#277]
Consider the app

```python
from cyclopts import App, Parameter
from typing import Annotated, Literal

app = App(help="Hello World", help_format=None)

@app.command
def greet(
    title: Annotated[str, Parameter(help="The title to use")],
    name: Annotated[str, Parameter(help="The name to greet")] = "World",
    formality: Annotated[Literal["formal", "informal"], Parameter(help="The formality to use")] = "formal",
):
    if formality == "informal":
        print(f"Hi {title} {name}")
    else:
        print(f"Hello {title} {name}")

app()
```

Using `help_format=None`
```bash
Usage: cli.py greet [ARGS] [OPTIONS]

╭─ Parameters ────────────────────────────────────────────────────────────────╮
│ *  TITLE --title          The title to use [required]                       │
│    NAME --name            The name to greet [default: World]                │
│    FORMALITY --formality  The formality to use [choices: formal, informal]  │
│                           [default: formal]                                 │
╰─────────────────────────────────────────────────────────────────────────────╯
```

Using `help_format="rich"`
```bash
Usage: cli.py greet [ARGS] [OPTIONS]

╭─ Parameters ────────────────────────────────────────────────────────────────╮
│ *  TITLE --title          The title to use                                  │
│    NAME --name            The name to greet                                 │
│    FORMALITY --formality  The formality to use                              │
╰─────────────────────────────────────────────────────────────────────────────╯
```

It seems like the square brackets are getting treated as rich-markup. 

It can be solved by escaping the markup, but I suppose there's a cleaner way

```cyclopts/help.py
...
def create_parameter_help_panel()
     ...

        if argument.parameter.show_choices:
            choices = _get_choices(argument.hint, argument.parameter.name_transform)
            if choices:
                if format == "rich":
                    help_text = rf"\[choices: {choices}]"
                else:
                    help_text = rf"[choices: {choices}]"
                help_append(help_text, "dim")

...
```

**Comment by BrianPugh** [comment]
thanks for the report! I'll make a PR shortly!

**Comment by BrianPugh** [comment]
resolved via #278 and will be released in v3.1.3 later today.
